### PR TITLE
librewolf-unwrapped: add darwin

### PR DESCRIPTION
**Pull request by uri-canva**
###### Description of changes

Add darwin support. Since upstream doesn't provide a darwin native build, but only a cross compile from Linux, use the prebuilt binaries.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
I think it should be added as a separate package.

**Comment by NickCao** [comment]
> Like `librewolf-bin` since it's prebuilt?

Yes, maybe even reuse the existing packaging infrastructure for firefox-bin.

File: pkgs/applications/networking/browsers/librewolf/src.json [hunk 1]
```diff
@@ -7,5 +7,17 @@
...
+  },
+  "bin": {
+    "mac": {
+      "x86_64": {
+        "url": "https://gitlab.com/librewolf-community/browser/macos/uploads/62a3836df78e94c13455d85abaf521c1/librewolf-113.0.2-1.en-US.mac.x86_64.dmg",
+        "sha256": "0sncfvxfmrs7ikv03m8xgd0ha58q8j508pf68x2yqkqnyr8anl0w"
+      },
+      "aarch64": {
+        "url": "https://gitlab.com/librewolf-community/browser/macos/uploads/ee6be51c4b9eee802cd7df889bfcf622/librewolf-113.0.2-1.en-US.mac.aarch64.dmg",
+        "sha256": "1p255g72cjssfsrbnbgfvakdz46p1gi0rpjah7w19b1kvgskx5j2"
```
Comment (skykanin) [inline pkgs/applications/networking/browsers/librewolf/src.json:19]: `https://gitlab.com/librewolf-community` is now archived use the [bsys6 repository release assets](https://gitlab.com/librewolf-community/browser/bsys6/-/releases) instead.

File: pkgs/applications/networking/browsers/librewolf/update.nix [hunk 2]
```diff
@@ -44,6 +45,13 @@ writeScript "update-librewolf" ''
     exit 1
   fi
 
+  macosAssets=$(curl https://gitlab.com/api/v4/projects/librewolf-community%2Fbrowser%2Fmacos/releases/v$latestTag/assets/links)
```
Comment (skykanin) [inline pkgs/applications/networking/browsers/librewolf/update.nix:48]: As previously mentioned `librewolf-community` is archived, use `https://gitlab.com/librewolf-community/browser/bsys6/-/archive/$latestTag-1/bsys6-$latestTag-1.tar.gz` instead

**Review by skykanin — CHANGES_REQUESTED** [review]
Some updated need to be made, but otherwise looks good

**Comment by skykanin** [comment]
> The existing `firefox-bin` only supports linux, but let me have a look, I'll try to follow the same pattern at least.

This shouldn't matter if you get the librewolf binaries directly from their [bsys6](https://gitlab.com/librewolf-community/browser/bsys6/-/releases) repository.

**Pull request closed without merging**